### PR TITLE
feat: moved jobs to steps, osv, semgrep enhancements

### PR DESCRIPTION
**Pull request by aakashdeshpandesg**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -85,30 +61,19 @@ on:
...
-        type: string
-        default: ""
+        default: ""        
 
 jobs:
   caching:
     name: Cache node modules
     runs-on: ubuntu-latest
     permissions: write-all
+    if: ${{!contains(github.actor, 'github-actions[bot]')  && (github.event_name != 'issue_comment') && (github.actor != 'dependabot[bot]') }}
```
Comment (mohankumar-studiographene) [inline .github/workflows/ci.yml]: github-actions[bot] check is happening twice in this `if` statement itself. Should be checked only once.
And combine all the checks into contains() func itself.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -178,27 +143,36 @@ jobs:
...
-    name: SAST Scan using semgrep
-    if: ${{ !contains( inputs.excluded_jobs, 'sast' ) && (github.actor != 'dependabot[bot]') }}
+
+
+  Security-scans:
+    name: Security scans
     runs-on: ubuntu-latest
     permissions: write-all
+    needs: caching
+    if: ${{!contains(github.actor, 'github-actions[bot]')  && (github.event_name != 'issue_comment') && (github.actor != 'dependabot[bot]') }}
```
Comment (mohankumar-studiographene) [inline .github/workflows/ci.yml]: github-actions[bot] check is happening twice in this `if` statement itself. Should be checked only once.
And combine all the checks into contains() func itself.

**Review by mohankumar-studiographene — CHANGES_REQUESTED** [review]
In all the jobs, github-actions[bot] check is happening twice in this `if` statement itself. Should be checked only once.
And combine all the checks into contains() func itself.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -227,333 +201,201 @@ jobs:
...
-      - name: Eslint scan
-        run: |
-          ${{ inputs.lint_command }}
-      - name: after step
-        if: ${{ inputs.lint_after_step_command != '' }}
-        run: |
-          ${{ inputs.lint_after_step_command }}
+      - name: Dependency Scan using Google OSV
+        id: dependency_scan
+        if: ${{ always() && !contains( inputs.excluded_jobs, 'dependency_scan' ) }}
```
Comment (mohankumar-studiographene) [inline .github/workflows/ci.yml:253]: aren't dependabot[bot], action bot, issue comment checks needed here?

**Review by mohankumar-studiographene — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (aakashdeshpandesg) [reply .github/workflows/ci.yml:253]: Actually that is the step, the main job is configured with the check for the bots

**Review by mohankumar-studiographene — APPROVED** [review]

**Pull request merged**